src/v/redpanda/admin_server.cc

```diff
@@ -670,6 +670,62 @@ static json_validator make_cluster_config_validator() {
     return json_validator(schema);
 }
 
+/**
+ * This function provides special case validation for configuration
+ * properties that need to check other properties' values as well
+ * as their own.
+ *
+ * Ideally this would be built into the config_store/property generic
+ * interfaces, but that's a lot of plumbing for a few relatively simple
+ * checks, so for the moment we just do the checks here by hand.
+ */
+void config_multi_property_validation(
+  ss::sstring const& username,
+  cluster::config_update_request const& req,
+  config::configuration const& updated_config,
+  std::map<ss::sstring, ss::sstring>& errors) {
```

Comment:
Should we avoid using std::map/unirdered_map/...

Reply:
I think it's okay for relatively small structures in command&control code (not the fast data path).  In this particular patch we're just passing a reference to an existing map, so there wasn't an active choice of type.